core/src/main/scala/fs2/internal/Future.scala

```diff
@@ -144,8 +144,8 @@ private[fs2] object Future {
 
   def suspend[A](f: => Future[A]): Future[A] = Suspend(() => f)
 
-  def async[A](listen: (A => Unit) => Unit): Future[A] =
-    Async((cb: A => Trampoline[Unit]) => listen { a => cb(a).run })
+  def async[A](listen: (A => Unit) => Unit)(implicit S: Strategy): Future[A] =
+    Async((cb: A => Trampoline[Unit]) => listen { a => S { cb(a).run } })
```

Comment:
+1 I think we perhaps have to visit the Async code, as some of the S() wrapped callbacks will be redundand now.